src/translator/cache.cpp

```diff
@@ -0,0 +1,168 @@
+#include "translator/cache.h"
+
+#include <cstdlib>
+
+namespace marian {
+namespace bergamot {
+
+#ifndef WASM_COMPATIBLE_SOURCE
+
+ThreadSafeL4Cache::ThreadSafeL4Cache(Ptr<Options> options)
+    : epochManagerConfig_(options->get<size_t>("cache-ebr-queue-size"),
+                          std::chrono::milliseconds(options->get<size_t>("cache-ebr-interval")),
+                          options->get<size_t>("cache-ebr-num-queues")),
+      cacheConfig_(options->get<size_t>("cache-size") * 1024 * 1024,
+                   std::chrono::seconds(options->get<size_t>("cache-time-to-live")),
+                   options->get<bool>("cache-remove-expired")),
+      service_(epochManagerConfig_),
+      context_(service_.GetContext()) {
+  // There is only a single cache we use. However, L4 construction API given by example demands we give it a string
+  // identifier.
+  const std::string cacheIdentifier = "global-cache";
```

Comment:
Does this mean there can only be one instance of this object?

Reply:
I would expect so. The following is the example provided by L4 to work with a Cache, which is adapted for our use-case:

* https://github.com/browsermt/L4/blob/master/Examples/main.cpp

The string might be to get the same instance in a multiproc setting.

Reply:
We definitely want this to be a singleton object.

Reply:
Wrong no need for singleton.